enhancements/proposals/1277-expose-scyllacluster/README.md

```diff
@@ -0,0 +1,398 @@
+type NodeListenOptions struct {
+    // nodes specifies options related to the address that is bound to socket used for communication with other nodes.
+    // This field controls the `listen_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"Any"}
+    Nodes ListenOptions `json:"nodes"`
+    
+    // clients specifies options related to the address that is bound to socket used for communication with clients.
+    // This field controls the `rpc_address` value in ScyllaDB config.
+    // +kubebuilder:default:={type:"Any"}
+    Clients ListenOptions `json:"clients"`
+}
+
+type NodeServiceType string
+
+const (
+    // HeadlessServiceType means nodes will be exposed via Headless Service.
+    HeadlessServiceType NodeServiceType = "Headless"
+    
+    // ClusterIPServiceType means nodes will be exposed via ClusterIP Service.
+    ClusterIPServiceType NodeServiceType = "ClusterIP"
+    
+    // LoadBalancerServiceType means nodes will be exposed via LoadBalancer Service.
+    LoadBalancerServiceType NodeServiceType = "LoadBalancer"
+)
+
+type NodeServiceTemplate struct {
+    // annotations is a custom key value map merged with every node Service annotations.
+    // +optional
+    Annotations map[string]string `json:"annotations,omitempty"`
+    
+    // type is the Kubernetes Service type.
+    // +kubebuilder:validation:Required
+    Type NodeServiceType `json:"type"`
+    
+    // externalTrafficPolicy controls value of corev1/Service.spec.externalTrafficPolicy of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    ExternalTrafficPolicy *corev1.ServiceExternalTrafficPolicy `json:"externalTrafficPolicy,omitempty"`
+    
+    // allocateLoadBalancerNodePorts controls value of corev1/Service.spec.allocateLoadBalancerNodePorts of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    AllocateLoadBalancerNodePorts *bool `json:"allocateLoadBalancerNodePorts,omitempty"`
+    
+    // loadBalancerClass controls value of corev1/Service.spec.loadBalancerClass of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    LoadBalancerClass *string `json:"loadBalancerClass,omitempty"`
+    
+    // internalTrafficPolicy controls value of corev1/Service.spec.internalTrafficPolicy of each node Service.
+    // Check Kubernetes corev1.Service documentation about semantic of this field.
+    // +optional
+    InternalTrafficPolicy *corev1.ServiceInternalTrafficPolicy `json:"internalTrafficPolicy,omitempty"`
+}
+
+// ExposeOptions hold options related to exposing ScyllaCluster backends.
+type ExposeOptions struct {
+    // nodeService controls properties of Service dedicated for each ScyllaCluster node.
+    // +kubebuilder:default:={type:"ClusterIP"}
+    NodeService *NodeServiceTemplate `json:"nodeService,omitempty"`
```

Comment:
Am I understanding correctly that it'd be possible to configure an SC to create services of type ClusterIP, but then choose e.g.  `ServiceLoadBalancerIngressIPBroadcastAddressType` for broadcast_address or broadcast_rpc_address? Would that be validated somehow? The proposal doesn't seem to mention it.

Reply:
Added validation as part of Risks and Mitigations